app/lib/replay/Problems.ts

```diff
@@ -48,86 +55,241 @@ export interface BoltProblem extends BoltProblemDescription {
   solution?: BoltProblemSolution;
 }
 
-export type BoltProblemInput = Omit<BoltProblem, "problemId" | "timestamp">;
+export type BoltProblemInput = Omit<BoltProblem, 'problemId' | 'timestamp'>;
+
+// Helper functions for mapping between Supabase and Bolt types
+function mapSupabaseToBoltStatus(status: Database['public']['Tables']['problems']['Row']['status']): BoltProblemStatus {
+  switch (status) {
+    case 'pending':
+      return BoltProblemStatus.Pending;
+    case 'solved':
+      return BoltProblemStatus.Solved;
+    case 'unsolved':
+      return BoltProblemStatus.Unsolved;
+    default:
+      return BoltProblemStatus.Pending;
+  }
+}
+
+function mapBoltToSupabaseStatus(
+  status?: BoltProblemStatus,
+): Database['public']['Tables']['problems']['Row']['status'] {
+  switch (status) {
+    case BoltProblemStatus.Pending:
+      return 'pending';
+    case BoltProblemStatus.Solved:
+      return 'solved';
+    case BoltProblemStatus.Unsolved:
+      return 'unsolved';
+    default:
+      return 'pending';
+  }
+}
+
+function convertSupabaseProblemToBolt(problem: Database['public']['Tables']['problems']['Row']): BoltProblem {
+  // Convert comments
+  const comments: BoltProblemComment[] = (problem.problem_comments || []).map((comment) => ({
+    username: comment.username,
+    content: comment.content,
+    timestamp: new Date(comment.created_at).getTime(),
+  }));
+
+  return {
+    version: 1,
+    problemId: problem.id,
+    timestamp: new Date(problem.created_at).getTime(),
+    title: problem.title,
+    description: problem.description,
+    status: mapSupabaseToBoltStatus(problem.status),
+    keywords: problem.keywords,
+    repositoryContents: problem.repository_contents as string,
+    comments,
+  };
+}
+
+function convertBoltProblemDescriptionToSupabase(
```

Comment:
Can we remove the converters and stick with the original naming scheme please? I don't want to migrate more than necessary.